UI components added and made the tag page look better

ImmortalSul opened the pull request.

ImmortalSul committed:
Adding components, hopefully this works

ImmortalSul committed:
fixing dependencies

jacksturt committed:
fix tls error

ImmortalSul committed:
Adding the tags components

ImmortalSul committed:
Fixing dependency issues + fix for mail

jacksturt commented on src/app/testing/tags/page.tsx:
```diff
@@ -13,38 +31,194 @@ export default function TestingPage() {
+          )}
+        </div>
       </div>
     </div>
```
So for this file, what I want to see pretty much is anything lowercase <div>, <label>, etc... is replaced by our custom components

ImmortalSul replied:
Noted.

ImmortalSul replied:
Done, let me know if you need anymore changes!

jacksturt reviewed (changes requested):
Can you please change this so that 
1. The example tags page is using all custom compoents
2. Add in the pages you had from your example files where all these components are named and on a page, so that people can open those page(s) for easy reference

ImmortalSul committed:
feat: improve theming and component library

- Fixed dark theme styling issues in dropdown menus and selects
- Removed DaisyUI in favor of custom styling
- Enhanced Tags page with better UX and custom components
- Added comprehensive component library pages with testing routes
- Fixed ESLint unescaped entity errors across components
- Added ScrollArea component for better content organization

jacksturt reviewed (approved):
Yes this is what I was hoping for good shit brother
